tools/plugin_template/pluma-plugin.c

```diff
@@ -25,160 +25,286 @@
 ##ifdef WITH_MENU
 /* UI string. See pluma-ui.xml for reference */
-static const gchar ui_str =
-	"<ui>"
-	"  <menubar name='MenuBar'>"
-	"    <!-- Put your menu entries here -->"
-	"  </menubar>"
-	"</ui>";
+const gchar submenu[] =
+"<ui>"
+"  <menubar name='MenuBar'>"
+"    <!-- Put your menu entries here -->"
+"  </menubar>"
+"</ui>";
 
 /* UI actions */
 static const GtkActionEntry action_entries[] =
 	{
 		/* Put your actions here */
 	};
-
-typedef struct
-{
-	GtkActionGroup *action_group;
-	guint           ui_id;
-} WindowData;
 ##endif
 
 static void
 ##(PLUGIN_ID.lower)_plugin_init (##(PLUGIN_ID.camel)Plugin *plugin)
 {
-	plugin->priv = ##(PLUGIN_ID.upper)_PLUGIN_GET_PRIVATE (plugin);
 
-	pluma_debug_message (DEBUG_PLUGINS,
-			     "##(PLUGIN_ID.camel)Plugin initializing");
+	plugin->priv = ##(PLUGIN_ID.lower)_plugin_get_instance_private (plugin);
+
+	pluma_debug_message (DEBUG_PLUGINS, "##(PLUGIN_ID.camel)Plugin initializing");
 }
 
 static void
 ##(PLUGIN_ID.lower)_plugin_finalize (GObject *object)
 {
-	pluma_debug_message (DEBUG_PLUGINS,
-			     "##(PLUGIN_ID.camel)Plugin finalizing");
+	##(PLUGIN_ID.camel)Plugin *plugin = ##(PLUGIN_ID.upper)_PLUGIN (object);
+
+	pluma_debug_message (DEBUG_PLUGINS, "##(PLUGIN_ID.camel)Plugin finalizing");
 
 	G_OBJECT_CLASS (##(PLUGIN_ID.lower)_plugin_parent_class)->finalize (object);
 }
 
-##ifdef WITH_MENU
 static void
-free_window_data (WindowData *data)
+##(PLUGIN_ID.lower)_plugin_dispose (GObject *object)
 {
-	g_return_if_fail (data != NULL);
+	##(PLUGIN_ID.camel)Plugin *plugin = ##(PLUGIN_ID.upper)_PLUGIN (object);
+
+	pluma_debug_message (DEBUG_PLUGINS, "##(PLUGIN_ID.camel)Plugin disposing");
+
+	if (plugin->priv->window != NULL)
+	{
+		g_object_unref (plugin->priv->window);
+		plugin->priv->window = NULL;
+	}
+
+	if (plugin->priv->action_group)
+	{
+		g_object_unref (plugin->priv->action_group);
+		plugin->priv->action_group = NULL;
+	}
 
-	g_object_unref (data->action_group);
-	g_free (data);
+	G_OBJECT_CLASS (##(PLUGIN_ID.lower)_plugin_parent_class)->dispose (object);
 }
+
+static void
+update_ui (##(PLUGIN_ID.camel)PluginPrivate *data)
+{
+##ifdef WITH_MENU
+	PlumaWindow *window;
+	PlumaView *view;
+	GtkAction *action;
+##endif
+
+	pluma_debug (DEBUG_PLUGINS);
+
+##ifdef WITH_MENU
+	window = PLUMA_WINDOW (data->window);
+	view = pluma_window_get_active_view (window);
+
+	pluma_debug_message (DEBUG_PLUGINS, "View: %p", view);
+
+	action = gtk_action_group_get_action (data->action_group,
+                                              "##(PLUGIN_ID.camel)PluginActions");
+	gtk_action_set_sensitive (action,
+                                  (view != NULL) &&
+                                  gtk_text_view_get_editable (GTK_TEXT_VIEW (view)));
 ##endif
+}
+
 
 static void
-impl_activate (PlumaPlugin *plugin,
-	       PlumaWindow *window)
+##(PLUGIN_ID.lower)_plugin_activate (PeasActivatable *activatable)
```

Comment:
Also this function needs to be updated, see https://github.com/mate-desktop/pluma/blob/f628a6ef38b7c42febf8fbc450d46ab1f5461f6f/plugins/trailsave/pluma-trail-save-plugin.c#L162 for example.